src/library/Connect/LocalEnvironmentManager.cs

```diff
@@ -436,12 +436,26 @@ private void _ReportProgress(EventLevel eventLevel, string message, params objec
 
                 // because we are using dns name instead of service we have to retrieve it by splitting when needed
                 // If this ever cause issues we should consider larger refactor where we add serviceName member variable to EndpointInfo class.
-                var serviceName = endpoint.DnsName
+                var dnsNameArray = endpoint.DnsName
                     .ToUpperInvariant()
-                    .Split(".")
+                    .Split(".");
+
+                var serviceName = dnsNameArray
                     .First()
-                    .Replace("-", "_")
-                    .Replace(".", "_");
+                    .Replace("-", "_");
+
+                // when !endpoint.IsInWorkloadNamespace
+                var serviceNs = dnsNameArray.Length == 2

```

Comment:
@garyburgmann Thanks for the PR, can you add some description to the PR why you need this change and if possible link to a GH issue. Thanks.